### PR TITLE
Handling zeros in readPerfStat

### LINKED ISSUE #2629: Wrong values for perf core events when starting the measurement
Hi,

I noticed wired behavior of metrics when using cAdvisor with perf + Prometheus + Grafana. I noticed, that if I use more core perf events that the platform has counters (which triggers event scaling), I get some spikes in the beginning of data collection for a given container. I did some investigation and I found [this function](https://github.com/google/cadvisor/blob/master/perf/collector_libpfm.go#L101).

In some cases, the reading returns Value 0, Time Running 0 and Time Enabled non 0. This leads to returning an undefined number from the function, by which I mean some crazy high value that breaks all the stats. I did some investigation on why this is happening. Only thing I was able to find in this context, is [this part of perf tool documentation](https://perf.wiki.kernel.org/index.php/Tutorial#multiplexing_and_scaling_events), which says in the context in event scaling:
```
This provides an estimate of what the count would have been, had the event been measured during the entire run. It is very
important to understand this is an estimate not an actual count. Depending on the workload, there will be blind spots
which can introduce errors during scaling.
```

This lead me to conclusion that we may be facing an issue, that for some reason, although PID can be measured on a core, for some reason it is not, thus we get non 0 time enabled, but no value or time running. This pretty much makes metrics corrupted(rate charts have some spikes that make them unreadable). 

My suggestion is to check if the value returned is equal to 0, and if it is, return PerfStat with all fields set to 0, as if nothing happened.

### PR DESCRIPTION
**Pull request by iwankgb**
Adds more robust handling for zeros in `readPerfStat` and fixes #2629 (hopefully).

### CODE REVIEW

**Comment by iwankgb** [comment]
While working on the PR I noticed following linter issues:
```
ubuntu@malyna4:~/go/src/github.com/google/cadvisor$ golangci-lint run --build-tags libpfm,cgo
WARN [linters context/goanalysis] Failed to get persisted facts: failed to get data from low-level cache by key fact_purity/facts for package yaml: failed to stat file /home/ubuntu/.cache/golangci-lint/4c/4c05768a80d95b176a37aae80ce01dbba0a393ee328bae71015e9bd10d999a71-d: stat /home/ubuntu/.cache/golangci-lint/4c/4c05768a80d95b176a37aae80ce01dbba0a393ee328bae71015e9bd10d999a71-d: no such file or directory
WARN [linters context/goanalysis] Failed to get persisted facts: failed to get data from low-level cache by key ctrlflow/facts for package yaml: failed to stat file /home/ubuntu/.cache/golangci-lint/68/684eaab543416e6ecf05612d3269cb5401637f7c9ab58a9e6d4ff3b84ad99df5-d: stat /home/ubuntu/.cache/golangci-lint/68/684eaab543416e6ecf05612d3269cb5401637f7c9ab58a9e6d4ff3b84ad99df5-d: no such file or directory
integration/tests/api/perf_test.go:50:28: Error return value of `framework.RetryForDuration` is not checked (errcheck)
	framework.RetryForDuration(func() error {
	                          ^
perf/uncore_libpfm.go:302:9: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
	} else {
	       ^
perf/uncore_libpfm.go:101:27: S1019: should use make(uncorePMUs) instead (gosimple)
	pmus := make(uncorePMUs, 0)
	                         ^
perf/uncore_libpfm_test.go:43:80: SA9002: file mode '777' evaluates to 01411; did you mean '0777'? (staticcheck)
	err = ioutil.WriteFile(filepath.Join(firstPMUPath, "cpumask"), []byte("0-1"), 777)
	                                                                              ^
perf/uncore_libpfm_test.go:47:76: SA9002: file mode '777' evaluates to 01411; did you mean '0777'? (staticcheck)
	err = ioutil.WriteFile(filepath.Join(firstPMUPath, "type"), []byte("18"), 777)
	                                                                          ^
perf/uncore_libpfm_test.go:58:81: SA9002: file mode '777' evaluates to 01411; did you mean '0777'? (staticcheck)
	err = ioutil.WriteFile(filepath.Join(secondPMUPath, "cpumask"), []byte("0,1"), 777)
	                                                                               ^
perf/uncore_libpfm_test.go:62:77: SA9002: file mode '777' evaluates to 01411; did you mean '0777'? (staticcheck)
	err = ioutil.WriteFile(filepath.Join(secondPMUPath, "type"), []byte("19"), 777)
```
I will fix them in separate commit.

**Comment by katarzyna-z** [comment]
@wacuuu Could you check this fix?

File: perf/collector_libpfm.go [hunk 1]
```diff
@@ -112,12 +112,17 @@ func readPerfStat(file readerCloser, name string, cpu int) (*info.PerfStat, erro
 	}
 
 	scalingRatio := 1.0
-	if perfData.TimeEnabled != 0 {
+	if perfData.TimeRunning != 0 && perfData.TimeEnabled != 0 {
```
Comment (katarzyna-z) [inline perf/collector_libpfm.go:115]: It seems that the same fix should be provided for uncore perf events, see this [line](https://github.com/google/cadvisor/blob/50b23f4ed9bc53cf068316b67bee04c4145f1e73/perf/uncore_libpfm.go#L379)

File: perf/collector_libpfm_test.go [hunk 2]
```diff
@@ -112,3 +112,134 @@ func TestNewCollector(t *testing.T) {
...
+			TimeRunning: 0,
+			ID:          0,
+		},
+		name: "some metric",
+		cpu:  4,
+		perfStat: info.PerfStat{
+			ScalingRatio: 1.0,
+			Value:        0,
+			Name:         "some metric",
+			Cpu:          4,
```
Comment (wacuuu) [inline perf/collector_libpfm_test.go:228]: For complicity:
missing the testcase that started whole of the issue:
`{
		Value:       0,
		TimeEnabled: 3,
		TimeRunning: 0,
	}`

**Comment by dashpole** [comment]
I'll leave this open for a day to let you fix the remaining comments above

**Comment by iwankgb** [comment]
@dashpole I think it can be merged :)

**Pull request merged**